src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker.cs

```diff
@@ -0,0 +1,499 @@
+                    var compilationOptions = commandLineArgs.CompilationOptions
+                        .WithXmlReferenceResolver(new XmlFileResolver(projectDirectory))
+                        .WithSourceReferenceResolver(new SourceFileResolver(ImmutableArray<string>.Empty, projectDirectory))
+                        // TODO: https://github.com/dotnet/roslyn/issues/4967
+                        .WithMetadataReferenceResolver(new WorkspaceMetadataFileReferenceResolver(metadataService, new RelativePathResolver(ImmutableArray<string>.Empty, projectDirectory)))
+                        .WithStrongNameProvider(new DesktopStrongNameProvider(commandLineArgs.KeyFileSearchPaths))
+                        .WithAssemblyIdentityComparer(DesktopAssemblyIdentityComparer.Default);
+
+                    var documents = CreateDocumentInfos(projectFileInfo.Documents, projectId, commandLineArgs.Encoding);
+                    var additionalDocuments = CreateDocumentInfos(projectFileInfo.AdditionalDocuments, projectId, commandLineArgs.Encoding);
+                    CheckForDuplicateDocuments(documents, additionalDocuments, projectPath, projectId);
+
+                    var analyzerReferences = ResolveAnalyzerReferences(commandLineArgs);
+
+                    var resolvedReferences = await ResolveReferencesAsync(projectId, projectFileInfo, commandLineArgs, cancellationToken).ConfigureAwait(false);
+
+                    return ProjectInfo.Create(
+                        projectId,
+                        version,
+                        projectName,
+                        assemblyName,
+                        language,
+                        projectPath,
+                        outputFilePath: projectFileInfo.OutputFilePath,
+                        outputRefFilePath: projectFileInfo.OutputRefFilePath,
+                        compilationOptions: compilationOptions,
+                        parseOptions: parseOptions,
+                        documents: documents,
+                        projectReferences: resolvedReferences.ProjectReferences,
+                        metadataReferences: resolvedReferences.MetadataReferences,
+                        analyzerReferences: analyzerReferences,
+                        additionalDocuments: additionalDocuments,
+                        isSubmission: false,
+                        hostObjectType: null);
+                });
+            }
+
+            private static VersionStamp GetProjectVersion(string projectFilePath)
+            {
+                if (!string.IsNullOrEmpty(projectFilePath) && File.Exists(projectFilePath))
+                {
+                    return VersionStamp.Create(File.GetLastWriteTimeUtc(projectFilePath));
+                }
+                else
+                {
+                    return VersionStamp.Create();
+                }
+            }
+
+            private static string GetAssemblyNameFromProjectPath(string projectFilePath)
```

Comment:
> private static string GetAssemblyNameFromProjectPath(string projectFilePath [](start = 11, length = 76)

This has the same input source to projectFilePath but doesn't consider the null / empty case.

Reply:
Fair point. :smile: